src/Avalonia.Base/Styling/ValueStyleQuery.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Linq;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Styling
+{
+    public abstract class ValueStyleQuery<T> : StyleQuery
```

Comment:
The class itself can be made internal - all inheritors are internal.

Reply:
Updated